File: protocol/go/references.go
package lse

import "strings"

// DefaultReferences is the built-in set of recognized reference names.
var DefaultReferences = map[string]bool{
	"me":     true,
	"you":    true,
	"it":     true,
	"result": true,
	"event":  true,
	"target": true,
	"body":   true,
}

// IsValidReference checks if a value is a valid reference name.
// The check is case-insensitive.
func IsValidReference(value string, refs map[string]bool) bool {
	if refs == nil {
		refs = DefaultReferences
	}
	return refs[strings.ToLower(value)]
}

File: protocol/typescript/src/references.ts
/** The built-in set of recognized reference names. */
export const DEFAULT_REFERENCES: ReadonlySet<string> = new Set([
  'me',
  'you',
  'it',
  'result',
  'event',
  'target',
  'body',
]);

/**
 * Checks if a value is a valid reference name.
 * The check is case-insensitive.
 */
export function isValidReference(
  value: string,
  refs: ReadonlySet<string> = DEFAULT_REFERENCES,
): boolean {
  return refs.has(value.toLowerCase());
}
